src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java

```diff
@@ -259,19 +260,26 @@ public MetricsLogger setTimestamp(Instant timestamp) throws InvalidTimestampExce
         return this;
     }
 
-    @SneakyThrows
     private void configureContextForEnvironment(MetricsContext context, Environment environment) {
         if (context.hasDefaultDimensions()) {
             return;
         }
         DimensionSet defaultDimension = new DimensionSet();
-        defaultDimension.addDimension("LogGroup", environment.getLogGroupName());
-        defaultDimension.addDimension("ServiceName", environment.getName());
-        defaultDimension.addDimension("ServiceType", environment.getType());
+        setDefaultDimension(defaultDimension, "LogGroup", environment.getLogGroupName());
+        setDefaultDimension(defaultDimension, "ServiceName", environment.getName());
+        setDefaultDimension(defaultDimension, "ServiceType", environment.getType());
         context.setDefaultDimensions(defaultDimension);
         environment.configureContext(context);
     }
 
+    private void setDefaultDimension(DimensionSet defaultDimension, String dimKey, String dimVal) {
+        try {
+            defaultDimension.addDimension(dimKey, dimVal);
+        } catch (InvalidDimensionException | DimensionSetExceededException e) {
+            // just do nothing
+        }
+    }
```

Comment:
To follow convention/readability I would suggest changing it to:

```suggestion
        try {
            defaultDimension.addDimension(dimKey, dimVal);
        } catch (InvalidDimensionException | DimensionSetExceededException ignored) {}
    }
```

Reply:
tried but not spotless style. not sure there's any difference here

Reply:
change the exception from `e` to `ignored`